### PR TITLE
Specify aggregatable report delivery

### PR DESCRIPTION
**Pull request by linnan-github**
***
<a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/pull/478.html" title="Last updated on Oct 17, 2022, 8:47 PM UTC (1d0258e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/478/737c5cd...1d0258e.html" title="Last updated on Oct 17, 2022, 8:47 PM UTC (1d0258e)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1387,33 +1388,72 @@ To <dfn>serialize an [=event-level report=]</dfn> |report|, run the following st
     [=serialize an integer|serialized=].
 1. Return the [=byte sequence=] resulting from executing [=serialize an infra value to JSON bytes=] on |data|.
 
+To <dfn>serialize an [=aggregatable report=] </dfn> |report|, run the following steps:
+
+1. Let |destination| be |report|'s [=aggregatable report/attribution destination=].
+1. [=Assert=]: |destination| is not an [=opaque origin=].
```
Comment (apasel422) [inline index.bs:1394]: Since we have this assertion for both event-level and aggregatable, can we move it to the general report serialization algorithm?

**Review by apasel422 — DISMISSED** [review]

**Review by apasel422 — APPROVED** [review]

File: index.bs [hunk 2]
```diff
@@ -1610,33 +1609,71 @@ To <dfn>serialize an [=event-level report=]</dfn> |report|, run the following st
     [=serialize an integer|serialized=].
 1. Return the [=byte sequence=] resulting from executing [=serialize an infra value to JSON bytes=] on |data|.
 
+To <dfn>serialize an [=aggregatable report=] </dfn> |report|, run the following steps:
```
Comment (apasel422) [inline index.bs:1612]:
```suggestion
To <dfn>serialize an [=aggregatable report=]</dfn> |report|, run the following steps:
```

File: index.bs [hunk 3]
```diff
@@ -1610,33 +1609,71 @@ To <dfn>serialize an [=event-level report=]</dfn> |report|, run the following st
...
+    |data|["`source_debug_key`"] to |report|'s [=aggregatable report/source debug key=],
+    [=serialize an integer|serialized=].
+1. If |report|'s [=aggregatable report/trigger debug key=] is not null, [=map/set=]
+    |data|["`trigger_debug_key`"] to |report|'s [=aggregatable report/trigger debug key=],
+    [=serialize an integer|serialized=].
+1. Return the [=byte sequence=] resulting from executing [=serialize an infra value to JSON bytes=] on |data|.
+
+Issue: Spec |data|["`shared_info`"] and |data|["`aggregation_service_payloads`"].
+
+To <dfn> serialize an [=attribution report=]</dfn> |report|, run the following steps:
```
Comment (apasel422) [inline index.bs:1631]:
```suggestion
To <dfn>serialize an [=attribution report=]</dfn> |report|, run the following steps:
```

File: index.bs [hunk 4]
```diff
@@ -1610,33 +1609,71 @@ To <dfn>serialize an [=event-level report=]</dfn> |report|, run the following st
...
+
+1. [=Assert=]: |report|'s [=attribution report/attribution destination=] is not an [=opaque origin=].
+1. If |report| is an:
+    <dl class="switch">
+    <dt>[=event-level report=]</dt>
+    <dd>Return the result of running [=serialize an event-level report=] with |report|.</dd>
+
+    <dt>[=aggregatable report=]</dt>
+    <dd>Return the result of running [=serialize an aggregatable report=] with |report|.</dd>
+    <dl>
```
Comment (apasel422) [inline index.bs:1641]:
```suggestion
    </dl>
```

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: index.bs [hunk 5]
```diff
@@ -1610,33 +1609,71 @@ To <dfn>serialize an [=event-level report=]</dfn> |report|, run the following st
...
+1. If |report| is an:
+    <dl class="switch">
+    <dt>[=event-level report=]</dt>
+    <dd>Return the result of running [=serialize an event-level report=] with |report|.</dd>
+
+    <dt>[=aggregatable report=]</dt>
+    <dd>Return the result of running [=serialize an aggregatable report=] with |report|.</dd>
+    </dl>
+
 Note: The inclusion of "`report_id`" in the report body is intended to allow the report recipient
```
Comment (apasel422) [inline index.bs:1643]: Is it possible to keep this note immediately below the `serialize an event-level report` dfn?

Comment (linnan-github) [reply index.bs:1643]: This note may apply to aggregatable reports as well to avoid double counting?

**Review by apasel422 — APPROVED** [review]

**Comment by apasel422** [comment]
LGTM

**Pull request merged**